### PR TITLE
4346 Enhance contribution guide for optional package

### LINKED ISSUE #4346: Enhance contribution guide for optional packages
**Is your feature request related to a problem? Please describe.**
Several of our new contributors asked duplicated questions about:
1. How to import other packages in the code, like skimage, torchvions, etc.
2. How to involve new optional packages in the codebase.

Would be nice to enhance the contribution guide doc for it.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #4346 .

### Description
According to the feedback from several recent new contributors, this PR added the description for optional package.
And also fixed a link error in `installation.md` doc.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
     1. [Unit testing](#unit-testing)
     1. [Building the documentation](#building-the-documentation)
     1. [Automatic code formatting](#automatic-code-formatting)
+    1. [Involve new package](#involve-new-package)
```
Comment (ericspod) [inline CONTRIBUTING.md:8]: Maybe a better title would be "Adding new dependencies"

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -171,6 +172,26 @@ The first line of the comment must be `/black` so that it will be interpreted by
...
+
+IgniteEngine, _ = optional_import("ignite.engine", 0.4.8, min_version, "Engine")
+
+
+class Workflow(IgniteEngine):
+    ...
+```
+If any of the newly added test program depends on the optional packages, please add it to the `exclude cases` of [min_tests](https://github.com/Project-MONAI/MONAI/blob/dev/tests/min_tests.py) to skip the minimal testing for it.
+
+To involve and config a new optional package in a pull request, please add the necessary information to below files:
```
Comment (ericspod) [inline CONTRIBUTING.md:190]: "To include and configure... " You might want to also explain very briefly what every file is meant for.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -171,6 +172,26 @@ The first line of the comment must be `/black` so that it will be interpreted by
...
+[setup.cfg](https://github.com/Project-MONAI/MONAI/blob/dev/setup.cfg), [docs/requirements.txt](https://github.com/Project-MONAI/MONAI/blob/dev/docs/requirements.txt), [requirements-dev.txt](https://github.com/Project-MONAI/MONAI/blob/dev/requirements-dev.txt), [environment-dev.yml](https://github.com/Project-MONAI/MONAI/blob/dev/environment-dev.yml), [installation.md](https://github.com/Project-MONAI/MONAI/blob/dev/docs/source/installation.md), [deviceconfig.py]https://github.com/Project-MONAI/MONAI/blob/dev/monai/config/deviceconfig.py.
```
Comment (wyli) [inline CONTRIBUTING.md:191]: missing brackets in the last url
```suggestion
[setup.cfg](https://github.com/Project-MONAI/MONAI/blob/dev/setup.cfg), [docs/requirements.txt](https://github.com/Project-MONAI/MONAI/blob/dev/docs/requirements.txt), [requirements-dev.txt](https://github.com/Project-MONAI/MONAI/blob/dev/requirements-dev.txt), [environment-dev.yml](https://github.com/Project-MONAI/MONAI/blob/dev/environment-dev.yml), [installation.md](https://github.com/Project-MONAI/MONAI/blob/dev/docs/source/installation.md), [deviceconfig.py](https://github.com/Project-MONAI/MONAI/blob/dev/monai/config/deviceconfig.py).
```

**Review by wyli — APPROVED** [review]
added some more info, please feel free to revise...

**Pull request merged**